extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/PolicyMappingConfig.java

```diff
@@ -67,4 +67,30 @@ public class PolicyMappingConfig {
      */
     @ConfigItem(defaultValue = "false")
     public boolean shared;
+
+    /**
+     * Whether permission check should be applied on all matching paths, or paths specific for the Jakarta REST resources.
+     */
+    @ConfigItem(defaultValue = "ALL")
+    public AppliesTo appliesTo;
+
+    /**
+     * Specifies additional criteria on paths that should be checked.
+     */
+    public enum AppliesTo {
+        /**
+         * Apply on all matching paths.
+         */
+        ALL,
+        /**
+         * Declares the permission check is only applied on the Jakarta REST request paths.
+         * This option must only be used in cases when authentication must happen after Jakarta REST endpoint
```

Comment:
```suggestion
         * Use this option to delay the permission check if an authentication mechanism is chosen with an annotation on the matching Jakarta REST endpoint.
```

Not sure it reads great but I'm trying to clarify somehow why exactly this option must be selected

Reply:
Yeah I get it. I tried to avoid delay for negative connotations, but now I can see it's fine. Let's use your suggestion.